tests/integration/workflows/test_workflow_actions.py

```diff
@@ -329,3 +399,53 @@ def test_normalize_journal_titles_unknown_journals_no_ref(workflow_app, insert_j
     assert obj.data['publication_info'][2]['journal_title'] == 'Unknown2'
     assert 'journal_record' not in obj.data['publication_info'][0]
     assert 'journal_record' not in obj.data['publication_info'][2]
+
+
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.actions.download_file_to_workflow',
+    side_effect=fake_download_file,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.arxiv.is_pdf_link',
+    return_value=True
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.beard.json_api_request',
+    side_effect=fake_beard_api_request,
+)
+@mock.patch(
+    'inspirehep.modules.workflows.tasks.magpie.json_api_request',
+    side_effect=fake_magpie_api_request,
+)
+def test_refextract_from_pdf(
+    mocked_api_request_magpie,
+    mocked_api_request_beard,
+    mocked_is_pdf_link,
+    mocked_package_download,
+    mocked_arxiv_download,
+    insert_cited_record,
+    workflow_app,
+    mocked_external_services
+):
+    """Test refextract by going through the entire workflow."""
+
+    citing_record, categories = insert_citing_record()
+
+    extra_config = {
+        "BEARD_API_URL": "http://example.com/beard",
+        "MAGPIE_API_URL": "http://example.com/magpie",
+        'ARXIV_CATEGORIES': categories,
+    }
+
+    with workflow_app.app_context():
```

Comment:
aren't you already in an application context by using the `workflow_app` fixture?

Reply:
I am not sure about this. Will check.

Reply:
It does work without this line, but I am unfamiliar with Flask so I am not completely sure what's going on.